### PR TITLE
Fix gather and scatter

### PR DESCRIPTION
**Pull request by okuta**
`link.params()` return order is unstable, So, this code sorts by parameter name.

### CODE REVIEW

File: tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py [hunk 1]
```diff
@@ -66,7 +66,6 @@ def test_gather_scatter_grads(self):
         t = chainer.Variable(chainer.cuda.to_gpu(t))
 
         loss0 = model0(x, t)
-        model1(x, t)
```
Comment (unnonouno) [inline tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: Why this line is removed?

Comment (okuta) [inline tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: This code has no effect.

File: chainer/training/updaters/multiprocess_parallel_updater.py [hunk 2]
```diff
@@ -307,7 +307,7 @@ def gather_grads(link):
     info = numpy.empty(num + 1, dtype=numpy.int32)
     info[0] = 0
     i = 0
-    for param in link.params():
+    for _, param in sorted(link.namedparams(), key=lambda x: x[0]):
```
Comment (niboshi) [inline chainer/training/updaters/multiprocess_parallel_updater.py]: Why not just `sorted(link.namedparams())`?

Comment (unnonouno) [inline chainer/training/updaters/multiprocess_parallel_updater.py]: I also think that is enough.

**Comment by okuta** [comment]
I fixed.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**